### PR TITLE
Create update branch

### PR DESCRIPTION
**Pull request by kayjeee**
I Add a column species of type string to the animal's table.

 Insert the all-new data.

 Update the animal's table by setting the species column to Digimon or pokimon.

 Delete all animals born after Jan 1st, 2022.

 Update all animals' weights that are negative to be their weight multiplied by -1.

 Write queries to answer all the asked questions.

Here are the screenshots 👍
 [take breath](https://drive.google.com/file/d/1r0CNZC5wwuv0_2h9UBFZRlkKueep1Y-0/view?usp=sharing)
[select from animals update ](https://drive.google.com/file/d/1D5Kzw-691ZMNxHF6DnmtREegJIixhRGE/view?usp=sharing)
[questions 2](https://drive.google.com/file/d/1GKqg8QdhgbgCzn3Ye3gtNHV2zy_kdHRB/view?usp=sharing)
[question 1](https://drive.google.com/file/d/1uc20jSXvhXisHg0ebTCChTkxlQWaZCA-/view?usp=sharing)
[commit the transaction](https://drive.google.com/file/d/1sF7612iUVYEpX-IfCvjd_DP6Kh50kJ9R/view?usp=sharing)
[inside the transaction 3](https://drive.google.com/file/d/1rQAv4ejN1qBvEwHc8I-zVAEnG_10E-v6/view?usp=sharing)
[inside the transaction 2](https://drive.google.com/file/d/1rJAx8Pm6lnBexkuNRVUCwrUNGnrrGhll/view?usp=sharing)
[inside the transaction ](https://drive.google.com/file/d/1801HpHgBMtvR1JQYZiYJDt-ovgJudPH1/view?usp=sharing)
[screenshot](https://drive.google.com/file/d/1ACJh21tSipEr-veFBq6Hcwn2PVpFXHBR/view?usp=sharing)

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -14,5 +14,6 @@ CREATE TABLE IF NOT EXISTS animals (
   date_of_birth DATE NOT NULL,
   escape_attempts INT NOT NULL,
   neutered BOOLEAN,
-  weight_kg DECIMAL(5, 2) NOT NULL
+  weight_kg DECIMAL(5, 2) NOT NULL,
+  species VARCHAR(50)
```
Comment (tugsem) [inline schema.sql:18]: - [ ] Please use alter table for adding a new column to the table as per the requirement.

File: data.sql [hunk 2]
```diff
@@ -1,13 +1,83 @@
...
+-- Delete all records in the animals table
+DELETE FROM animals;
+
+-- Verify that records were deleted (optional)
+SELECT * FROM animals;
+
+ROLLBACK;
+
+-- Verify that all records in the animals table still exist (optional)
+SELECT * FROM animals;
```
Comment (tugsem) [inline data.sql]: - [ ] Please add these transactions and all your queries into the queries.sql file as per the requirement. Everything that is a query that helps to analyze data should be saved in the queries.sql file.
- [ ] Please add this transaction as per the requirement: 

image

- [ ] Please add these queries into the queries.sql and add screenshots of each result into your pull request description. Right now, your screenshots aren't correct:
image

File: data.sql [hunk 3]
```diff
@@ -1,13 +1,83 @@
...
+
+ROLLBACK;
+
+-- Verify that the species column went back to the original state (optional)
+SELECT * FROM animals;
+
+-- Inside a transaction update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
+-- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+-- Verify that changes were made.
+COMMIT;
```
Comment (tugsem) [inline data.sql]: - [ ] Since you used rollback you don't have to use commit here, please remove it.

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello 👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️all required data is inserted
✔️the schema and data insertions added to the right file.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (kayjeee) [reply schema.sql:18]: done

Comment (kayjeee) [reply schema.sql:18]: done

File: data.sql (hunk 3 above)
Comment (kayjeee) [reply data.sql]: done

File: data.sql [hunk 4]
```diff
@@ -1,13 +1,112 @@
...
+SELECT neutered, MAX(escape_attempts) AS max_escape_attempts FROM animals GROUP BY neutered;
+
+-- What is the minimum and maximum weight of each type of animal?
+SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight FROM animals GROUP BY species;
+
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT species, AVG(escape_attempts) AS average_escape_attempts FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31' GROUP BY species;
+
+-- Rollback the transaction
+ROLLBACK;
```
Comment (Zuhaib042) [inline data.sql]: - [ ] As suggested by the previous reviewer, kindly consider removing all that is related to `queries` from data.sql file as per requirements.

File: queries.sql [hunk 5]
```diff
@@ -29,3 +27,80 @@ SELECT * FROM animals WHERE name <> 'Gabumon';
...
+DELETE FROM animals;
+
+-- Verify that records were deleted (optional)
+SELECT * FROM animals;
+
+ROLLBACK;
+
+-- Verify that all records in the animals table still exist (optional)
+SELECT * FROM animals;
+
```
Comment (Zuhaib042) [inline queries.sql]: - [ ] As suggested by the previous reviewer, kindly consider adding the following query that you missed:
  - Inside a transaction:
  - Delete all animals born after Jan 1st, 2022.
  - Create a savepoint for the transaction.
  - Update all animals' weight to be their weight multiplied by -1.
  - Rollback to the savepoint
  - Update all animals' weights that are negative to be their weight multiplied by -1.
  - Commit transaction

File: queries.sql [hunk 6]
```diff
@@ -29,3 +27,80 @@ SELECT * FROM animals WHERE name <> 'Gabumon';
...
+SELECT AVG(weight_kg) AS average_weight FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals?
+SELECT neutered, MAX(escape_attempts) AS max_escape_attempts FROM animals GROUP BY neutered;
+
+-- What is the minimum and maximum weight of each type of animal?
+SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight FROM animals GROUP BY species;
+
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT species, AVG(escape_attempts) AS average_escape_attempts FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31' GROUP BY species;
```
Comment (Zuhaib042) [inline queries.sql]: - [ ] As suggested by the previous reviewer, kindly consider adding the screenshots for these queries in your pull request.

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @kayjeee 
Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✅ Nice try implementing the suggested changes

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 5 above)
Comment (kayjeee) [reply queries.sql]: done

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @kayjeee , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  All requested changes implemented. 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**